Flaky `TestDatabricksWithGlueMetastoreCleanUp.testCleanUpOldTablesUsingDelta`

```
tests               | 2022-08-27 15:27:55 INFO: FAILURE     /    io.trino.tests.product.deltalake.TestDatabricksWithGlueMetastoreCleanUp.testCleanUpOldTablesUsingDelta (Groups: profile_specific_tests, delta-lake-databricks) took 10.6 seconds
tests               | 2022-08-27 15:27:55 SEVERE: Failure cause:
tests               | io.trino.tempto.query.QueryExecutionException: java.sql.SQLException: Query failed (#20220827_094255_00005_kvtct): line 1:1: Schema 'test_extraordinary_b91217y9yv0v' does not exist
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:119)
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.executeQuery(JdbcQueryExecutor.java:84)
tests               | 	at io.trino.tests.product.utils.QueryExecutors$1.lambda$executeQuery$0(QueryExecutors.java:60)
tests               | 	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
tests               | 	at net.jodah.failsafe.RetryPolicyExecutor.lambda$supply$0(RetryPolicyExecutor.java:62)
tests               | 	at net.jodah.failsafe.Execution.executeSync(Execution.java:129)
tests               | 	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
tests               | 	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:67)
tests               | 	at io.trino.tests.product.utils.QueryExecutors$1.executeQuery(QueryExecutors.java:60)
tests               | 	at io.trino.tests.product.deltalake.TestDatabricksWithGlueMetastoreCleanUp.cleanSchema(TestDatabricksWithGlueMetastoreCleanUp.java:92)
tests               | 	at io.trino.tests.product.deltalake.TestDatabricksWithGlueMetastoreCleanUp.lambda$testCleanUpOldTablesUsingDelta$2(TestDatabricksWithGlueMetastoreCleanUp.java:61)
tests               | 	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
tests               | 	at io.trino.tests.product.deltalake.TestDatabricksWithGlueMetastoreCleanUp.testCleanUpOldTablesUsingDelta(TestDatabricksWithGlueMetastoreCleanUp.java:61)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
tests               | Caused by: java.sql.SQLException: Query failed (#20220827_094255_00005_kvtct): line 1:1: Schema 'test_extraordinary_b91217y9yv0v' does not exist
tests               | 	at io.trino.jdbc.AbstractTrinoResultSet.resultsException(AbstractTrinoResultSet.java:1937)
tests               | 	at io.trino.jdbc.TrinoResultSet.getColumns(TrinoResultSet.java:285)
tests               | 	at io.trino.jdbc.TrinoResultSet.create(TrinoResultSet.java:61)
tests               | 	at io.trino.jdbc.TrinoStatement.internalExecute(TrinoStatement.java:262)
tests               | 	at io.trino.jdbc.TrinoStatement.execute(TrinoStatement.java:240)
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:128)
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:112)
tests               | 	... 25 more
tests               | 	Suppressed: java.lang.Exception: Query: SHOW TABLES IN test_extraordinary_b91217y9yv0v
tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:136)
tests               | 		... 26 more
tests               | Caused by: io.trino.spi.TrinoException: line 1:1: Schema 'test_extraordinary_b91217y9yv0v' does not exist
tests               | 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
tests               | 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
tests               | 	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:277)
tests               | 	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:216)
tests               | 	at io.trino.sql.tree.ShowTables.accept(ShowTables.java:72)
tests               | 	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
tests               | 	at io.trino.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java:213)
tests               | 	at io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:53)
tests               | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:76)
tests               | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:71)
tests               | 	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:261)
tests               | 	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:199)
tests               | 	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:825)
tests               | 	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:135)
tests               | 	at io.trino.$gen.Trino_393_144_g7a3c0f6____20220827_094216_2.call(Unknown Source)
tests               | 	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
tests               | 	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
tests               | 	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
```

https://github.com/trinodb/trino/runs/8049378719

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDatabricksWithGlueMetastoreCleanUp.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDatabricksWithGlueMetastoreCleanUp.java
@@ -26,15 +26,16 @@ import org.testng.annotations.Test;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
 import java.util.List;
-import java.util.Locale;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
 import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
 import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
 import static io.trino.tests.product.utils.QueryExecutors.onTrino;
 import static java.lang.String.format;
 import static java.lang.System.currentTimeMillis;
+import static java.util.Locale.ENGLISH;
 import static java.util.concurrent.TimeUnit.MINUTES;
 import static java.util.stream.Collectors.toUnmodifiableList;
 
@@ -53,7 +54,7 @@ public class TestDatabricksWithGlueMetastoreCleanUp
         List<String> schemas = onTrino().executeQuery("SELECT DISTINCT(table_schema) FROM information_schema.tables")
                 .rows().stream()
                 .map(row -> (String) row.get(0))
-                .filter(schema -> schema.toLowerCase(Locale.ROOT).startsWith("test") || schema.equals("default"))
+                .filter(schema -> schema.toLowerCase(ENGLISH).startsWith("test") || schema.equals("default"))
                 .collect(toUnmodifiableList());
 
         // this is needed to make deletion of some views possible
@@ -63,9 +64,12 @@ public class TestDatabricksWithGlueMetastoreCleanUp
 
     private void cleanSchema(String schema, long startTime, AWSGlueAsync glueClient)
     {
-        Set<String> allTableNames = findAllTestTablesInSchema(schema);
+        Set<String> allTestTableNames = findAllTablesInSchema(schema).stream()
+                .filter(name -> name.toLowerCase(ENGLISH).startsWith("test"))
+                .collect(toImmutableSet());
+        log.info("Found %d tables to drop in schema %s", allTestTableNames.size(), schema);
         int droppedTablesCount = 0;
-        for (String tableName : allTableNames) {
+        for (String tableName : allTestTableNames) {
             try {
                 Table table = glueClient.getTable(new GetTableRequest().withDatabaseName(schema).withName(tableName)).getTable();
                 Instant createTime = table.getCreateTime().toInstant();
@@ -89,7 +93,7 @@ public class TestDatabricksWithGlueMetastoreCleanUp
             }
         }
         log.info("Dropped %d tables in schema %s", droppedTablesCount, schema);
-        if (!schema.equals("default") && onTrino().executeQuery(format("SHOW TABLES IN %s", schema)).getRowsCount() == 0) {
+        if (!schema.equals("default") && findAllTablesInSchema(schema).isEmpty()) {
             try {
                 onTrino().executeQuery("DROP SCHEMA IF EXISTS " + schema);
                 log.info("Dropped schema %s", schema);
@@ -100,14 +104,12 @@ public class TestDatabricksWithGlueMetastoreCleanUp
         }
     }
 
-    private Set<String> findAllTestTablesInSchema(String schema)
+    private Set<String> findAllTablesInSchema(String schema)
     {
         try {
-            QueryResult allTables = onTrino().executeQuery(format("SHOW TABLES IN %s", schema));
-            log.info("Found %d potential tables to drop in schema %s", allTables.rows().size(), schema);
+            QueryResult allTables = onTrino().executeQuery(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", schema));
             return allTables.rows().stream()
                     .map(row -> (String) row.get(0))
-                    .filter(name -> name.toLowerCase(Locale.ROOT).startsWith("test"))
                     .collect(Collectors.toUnmodifiableSet());
         }
         catch (Exception e) {

